src/averageAges.js

```diff
@@ -52,7 +70,15 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => !onlyWithSon
+    ? people.find(mother => person.mother === mother.name)
+    : people.find(mother => person.mother === mother.name && person.sex === 'm')
+  );
+
+  const ageDiff = children.map(person => person.born
+    - people.find(mother => person.mother === mother.name).born);
```

Comment:
```suggestion
  const ageDiff = children.map(
    person => person.born - people.find(mother => person.mother === mother.name)
    .born
  );
```
Не разрываем коллбек. А вот разные методы, идущие друг за другом, лучше наоборот писать каждый с новой строки